include/oneapi/dpl/pstl/hetero/dpcpp/utils_ranges_sycl.h

```diff
@@ -538,30 +571,81 @@ struct __get_sycl_range
                                                                  __offset /* offset*/, __n /* size*/)};
     }
 
-    //specialization for a host iterator
+    //SFINAE-overload for a contiguous host iterator
+    template <typename _Iter>
+    auto
+    operator()(_Iter __first, _Iter __last)
+        -> ::std::enable_if_t<is_temp_buff<_Iter>::value && __is_addressable_v<_Iter> && !is_zip<_Iter>::value &&
+        !is_permutation<_Iter>::value, __range_holder<oneapi::dpl::__ranges::all_view<val_t<_Iter>, AccMode>>>
+    {
+        static_assert(__is_addressable_v<_Iter>);
```

Comment:
This `static_assert` looks redundant based on what I can see in `enable_if`

Reply:
it is just  for check that we have not a mistake in signature with enable_if. I think it is helpful.

Reply:
As we discussed I don't see how it is different from the rest of checks.

Reply:
removed